### PR TITLE
docs: Improve generation of documentation

### PR DESCRIPTION
**Pull request by ThisIsMissEm**
Replays the latest docs generation changes from `solid-client-js` on this repo.

For comparison, the current docs: https://solid-client-access-grants-js.vercel.app/modules/type_IssueAccessRequestParameters.html

# Checklist

- [x] All acceptance criteria are met.
- [x] Relevant documentation, if any, has been written/updated.
- [ ] The changelog has been updated, if applicable.
- [ ] New functions/types have been exported in `index.ts`, if applicable.
- [ ] New modules (i.e. new `.ts` files) are listed in the `exports` field in `package.json`, if applicable.
- [ ] New modules (i.e. new `.ts` files) are listed in the `typedocOptions.entryPoints` field in `tsconfig.json`, if applicable.
- [ ] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

**Comment by ThisIsMissEm** [comment]
@NSeydoux @matthieubosquet this isn't necessarily "done", but it would be good to get your eyes on it, as there are some code changes (some object types becoming concrete interfaces)

File: docs/api/Makefile [hunk 1]
```diff
@@ -17,56 +17,45 @@ ifeq ($(MY_OS),Darwin)
...
 
-SPHINXBUILD_AVAILABLE := $(shell command -v $(SPHINXBUILD) 2> /dev/null)
+.PHONY: Makefile help check clean clean-all prepare html dist
 
-all:
-	ifndef SPHINXBUILD_AVAILABLE
-		$(error sphinx-build is not available, please follow the instructions in ./docs/api/README.md)
-	endif
+check:
+  $(if $(shell command -v $(SPHINXBUILD) 2> /dev/null),$(info Found `$(SPHINXBUILD)`),$(error sphinx-build is not available, please follow the instructions in ./docs/api/README.md))
```
Comment (NSeydoux) [inline docs/api/Makefile:23]:
```suggestion
  $(if $(shell command -v $(SPHINXBUILD) 2> /dev/null),$(info Found `$(SPHINXBUILD)`),$(error sphinx-build is not available, please follow the instructions in ./docs/README.md))
```

File: docs/api/source/index.rst [hunk 2]
```diff
@@ -1,13 +1,102 @@
...
-solid-client-access-grants API
-==============================
+====================================================
+@inrupt/solid-client-access-grants API Documentation
+====================================================
 
+``@inrupt/solid-client-access-grants`` is a JavaScript library for
+requesting and managing access given to an agent for a resource. These
+access grants are signed by a Solid Pod owner’s private key, which
+itself is managed by the Solid Pod server.
```
Comment (NSeydoux) [inline docs/api/source/index.rst:10]:
```suggestion
requesting and managing access given to an agent for a resource. These
access grants are represented by Verifiable Credentials, signed by an Issuer associated to the Pod server where the resources are hosted.
```
I'm afraid the notion of Pod owner could be confusing here: the private key is managed by the Pod server's operator, and not really associated to the resource owner.

File: docs/api/source/index.rst [hunk 3]
```diff
@@ -1,13 +1,102 @@
...
+features that will work in all commonly-used browsers, except Internet
+Explorer. If you need support for Internet Explorer, it is recommended
+to pass them through a tool like `Babel <https://babeljs.io>`__, and to
+add polyfills for e.g. ``Map``, ``Set``, ``Promise``, ``Headers``,
+``Array.prototype.includes``, ``Object.entries`` and
+``String.prototype.endsWith``.
+
+Additionally, when using this package in an environment other than Node.js, you
+will need `a polyfill for Node's buffer module
+<https://www.npmjs.com/package/buffer>`__.
```
Comment (NSeydoux) [inline docs/api/source/index.rst:52]: Is this the case for this library too, or is it only for solid-client-authn ?

File: docs/api/Makefile (hunk 1 above)
Comment (ThisIsMissEm) [reply docs/api/Makefile:23]: Whoops! Good catch!

File: docs/api/source/index.rst (hunk 2 above)
Comment (ThisIsMissEm) [reply docs/api/source/index.rst:10]: oh! Good call, we need to update the README.md too then!

File: docs/api/source/index.rst (hunk 3 above)
Comment (ThisIsMissEm) [reply docs/api/source/index.rst:52]: I'm not certain, tbh, so I'm kinda just going with "standardise this across projects" for now

File: src/type/AccessBaseOptions.ts [hunk 4]
```diff
@@ -19,13 +19,17 @@
 // SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 //
 
+/**
+ * @module interfaces
+ */
```
Comment (ThisIsMissEm) [inline src/type/AccessBaseOptions.ts:24]: An open question @NSeydoux & @matthieubosquet is: Do we want the interfaces to be located in `src/type/` or would we be better with them in `src/interfaces/` ?

**Review by NSeydoux — APPROVED** [review]

File: src/type/index.ts [hunk 5]
```diff
@@ -22,3 +22,4 @@
 export { FetchOptions } from "./FetchOptions";
 export { RedirectOptions } from "./RedirectOptions";
 export { AccessBaseOptions } from "./AccessBaseOptions";
+export { AccessModes } from "./AccessModes";
```
Comment (ThisIsMissEm) [inline src/type/index.ts:25]: Aside: I think this should probably all be `export type` declarations instead maybe? I don't really know if there's that much of a difference though.

**Review by NSeydoux — APPROVED** [review]

File: src/type/AccessBaseOptions.ts (hunk 4 above)
Comment (NSeydoux) [reply src/type/AccessBaseOptions.ts:24]: I don't have a strong opinion, in fact I'm not even sure having them in a separate module is what we want. Do you have a preference between the two options ?

**Comment by NSeydoux** [comment]
I went ahead and merged because the e2e failures are unrelated

**Pull request merged**